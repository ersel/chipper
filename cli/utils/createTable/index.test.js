const createHTMLTable = require('./');

describe('HTML Table', () => {
	it('should generate HTML table from array of arrays', () => {
		const sampleResult = [
			{
				sourceFile: '/Users/erselaker/chipper/src/loginService.js',
				importedModules: [
					{
						source: './utils/cookiesService',
						absolute:
							'/Users/erselaker/chipper/src/utils/cookiesService.js',
						imports: [{ imported: 'setCookie', local: 'setCookie' }]
					}
				]
			}
		];

		expect(createHTMLTable(sampleResult)).toMatchSnapshot();
	});
});
