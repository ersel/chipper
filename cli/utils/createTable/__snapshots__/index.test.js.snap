// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HTML Table should generate HTML table from array of arrays 1`] = `
"
	<html>
	<head>
	<title>Chipper Scan Results</title>
	<style>
	
	table.blueTable {
		border: 1px solid #1C6EA4;
		background-color: #EEEEEE;
		width: 100%;
		text-align: left;
		border-collapse: collapse;
	  }
	  table.blueTable td, table.blueTable th {
		border: 1px solid #AAAAAA;
		padding: 3px 2px;
	  }
	  table.blueTable tbody td {
		font-size: 18px;
	  }
	  table.blueTable tr:nth-child(even) {
		background: #D0E4F5;
	  }
	  table.blueTable thead {
		background: #1C6EA4;
		background: -moz-linear-gradient(top, #5592bb 0%, #327cad 66%, #1C6EA4 100%);
		background: -webkit-linear-gradient(top, #5592bb 0%, #327cad 66%, #1C6EA4 100%);
		background: linear-gradient(to bottom, #5592bb 0%, #327cad 66%, #1C6EA4 100%);
		border-bottom: 2px solid #444444;
	  }
	  table.blueTable thead th {
		font-size: 15px;
		font-weight: bold;
		color: #FFFFFF;
		border-left: 2px solid #D0E4F5;
	  }
	  table.blueTable thead th:first-child {
		border-left: none;
	  }
	
	</style>
	</head>
	<table class=\\"blueTable\\">
	<thead>
	<tr>
	<th>Source Module</th><th>Imported Module</th><th>Imported Symbols</th></tr></thead><tr><td>/Users/erselaker/chipper/src/loginService.js</td><td>/Users/erselaker/chipper/src/utils/cookiesService.js</td><td>setCookie</td></tr></table></html>"
`;
